packages/hooks/src/hooks/useContract.ts

```diff
@@ -1,27 +1,38 @@
 import React from 'react';
 import { Web3Context } from '../Provider';
-import { Contract } from 'ethers';
+import { Contract, ContractInterface } from 'ethers';
 
-export function useContract(address: string, abi) {
+/**
+ * @description
+ * Defines the contract instance on `useState` hook
+ */
+export type ContractInstance<T extends Contract> = T | null | undefined;
+
+/**
+ * @description
+ * The return type of the `useContract` hook
+ */
+export type UseContractHook<T extends Contract> = [
+  T | null | undefined,
+  boolean
+];
+
+export function useContract<T extends Contract>(
+  address: string,
+  abi: ContractInterface
+): UseContractHook<T> {
   const context = React.useContext(Web3Context);
-  const [contract, setContract] = React.useState({});
+  const [contract, setContract] = React.useState<ContractInstance<T>>(null);
```

Comment:
When can this be `undefined` we're initializing it as null shouldn't we keep it stable as `T | null`?

Reply:
Ahh right, yeah it makes sense to remove `undefined` since it'll never be `undefined`